src/main/java/org/jetbrains/research/groups/ml_methods/refactoring/detection/RefactoringDetectionApplication.java

```diff
@@ -1,11 +1,77 @@
 package org.jetbrains.research.groups.ml_methods.refactoring.detection;
 
+import org.jetbrains.annotations.NotNull;
+
+import java.io.IOException;
+import java.net.URL;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.List;
+
 public class RefactoringDetectionApplication {
-    public static void main(String[] args) {
-        printUsage();
+    private static final String DEFAULT_DETECTION_TOOL_NAME = "RMiner";
+
+    public static void main(@NotNull String[] args) {
+        if (args.length != 3 && args.length != 2) {
+            printUsage();
+            return;
+        }
+
+        Path pathToRepositoriesFile = Paths.get(args[0]);
+        Path outputDirPath = Paths.get(args[1]);
+        String detectionToolName = args.length == 3 ? args[2] : DEFAULT_DETECTION_TOOL_NAME;
+        RefactoringDetectionTool refactoringDetectionTool;
+        switch (detectionToolName) {
+            case "RMiner":
+                refactoringDetectionTool = RefactoringDetectionToolFactory.createRMiner();
+                break;
+            default:
+                System.out.println("No such tool. Try again.");
+                printUsage();
+                return;
+        }
+        List<URL> repositories;
+        try {
+            repositories = RepositoriesReader.read(pathToRepositoriesFile);
+        } catch (IOException e) {
+            System.err.println("Error occurred during parsing passed repository urls.");
+            System.err.println("Reason: " + e.getMessage());
+            System.err.println("Application terminated.");
+            return;
+        }
+        List<DetectedRefactoringsInRepository> detectedRefactorings;
+        try {
+            detectedRefactorings = refactoringDetectionTool.detect(repositories);
+        } catch (Exception e) {
+            System.err.println("Error occurred during refactoring detection.");
+            System.err.println("Reason: " + e.getMessage());
+            System.err.println("Application terminated.");
+            return;
+        }
+        for (DetectedRefactoringsInRepository detectedRefactoringsInRepository : detectedRefactorings) {
+            String projectName = ParsingUtils.getProjectName(detectedRefactoringsInRepository.getRepository());
+            Path outputFilePath = outputDirPath.resolve(projectName);
+            try {
+                detectedRefactoringsInRepository.write(outputFilePath);
+            } catch (IOException e) {
+                System.err.println("Error occurred during writing to " + outputFilePath + " file.");
+                System.err.println("Reason: " + e.getMessage());
+                System.err.println("Refactorings of " + projectName + " project can be corrupted.");
+            }
+        }
+        System.out.println("====================STATISTIC====================");
```

Comment:
Useful values: number of processed commits, number of commits with refactorings, median of number of refactorings in commits (only for commits that contain refactorings), max number of refactorings in one commit.
Same values for all repositories together.

Reply:
Fixed.